src/Microsoft.Restier.Core/ApiBuilder.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Restier.Core
+{
+    public sealed class ApiBuilder
```

Comment:
Sorry I brought back ApiBuilder, but it's completely different now. It is to organize the configuration in a highly flexible way.
Now configuration is split into **inner **and **outer **parts, and consumer could add configuration to all the 4 ends.